### DISCUSSION TITLE
Load multiple collated data files into a big one

### DISCUSSION THREAD

**Discussion by mapapin** [pyg-team/pytorch_geometric#4781] [Q&A] [answered]
Hi, I want to load data from pre-computed pytorch files saved with `torch.save((data, slices), file_name)` and (data, slices) from PyG collate() into a big object (data, slices) which can be saved the same way.

I tried loading files and collate them again, but the slices are not correct.

Do you have a solution to collate collated datas ? 

Thanks !

**Comment by rusty1s** [comment, marked as answer]
```python
class MyDataset(InMemoryDataset):
    def __init__(self, data, slices):
        super().__init__(None)
        self.data, self.slices = data, slices


dataset1 = MyDataset(*torch.load(collated_data1_path))
dataset2 = MyDataset(*torch.load(collated_data1_path))

data_list = [d for d in dataset1] + [d for d in dataset2]
data, slices = InMemoryDataset.collate(data_list)
```

**Reply by mapapin** [reply]
Hi Matthias,
It works, thank you for being quick !

Have a nice day.
